carma-messenger-core/cpp_message/src/MobilityHeader_Message.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) 2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/**
+ * CPP File containing Mobility Header Message method implementations
+ */
+#include "MobilityHeader_Message.h"
+
+
+namespace cpp_message
+{
+    boost::optional<cav_msgs::MobilityHeader> Mobility_Header::decode_mobility_header_message(MobilityHeader_t *message_header){
+        cav_msgs::MobilityHeader header;
+        std::string sender_id, recipient_id, sender_bsm_id, plan_id, timestamp_string;
+        uint64_t timestamp;
+        auto str_len=message_header->hostStaticId.size;
+        if(str_len<=STATIC_ID_MAX_LENGTH && str_len!=0)
+        {
+            for(auto i=0;i<str_len;i++){
```

Comment:
Don't use auto in counter based for-loops. You can't know how the compiler will assign auto i=0. If it uses a uint8 or something you would get index exceptions. It's safer to use 
```
size_t i = 0;
```
For all your for loops that use a counter.